optimum/exporters/openvino/model_configs.py

```diff
@@ -4366,3 +4367,111 @@ def patch_model_for_export(
+                self.batch_size,
+                self.intermediate_size + 2 * self.mamba_ngroups * self.mamba_d_state,
+                self.conv_kernel_size,
+            )
+            conv_state = self.random_float_tensor(conv_state_shape, framework=framework, dtype=float_dtype)
+            kv_states_cache.append(conv_state)
+            ssm_state_shape = (self.batch_size, self.n_mamba_heads, self.mamba_headdim, self.ssm_state_size)
+            ssm_state = self.random_float_tensor(ssm_state_shape, framework=framework, dtype=float_dtype)
+            kv_states_cache.append(ssm_state)
+
+        return kv_states_cache
+
+
+@register_in_tasks_manager("zamba2", *["text-generation", "text-generation-with-past"], library_name="transformers")
+class Zamba2OpenVINOConfig(LlamaOpenVINOConfig):
+    PAD_ATTENTION_MASK_TO_PAST = False
+    DUMMY_INPUT_GENERATOR_CLASSES = (DummyTextInputGenerator, Zamba2DummyPastKeyValuesGenerator)
+    DUMMY_PKV_GENERATOR_CLASS = Zamba2DummyPastKeyValuesGenerator
+
+    def add_past_key_values_states(self, inputs_or_outputs: Dict[str, Dict[int, str]], direction: str):
+        if direction not in ["inputs", "outputs"]:
+            raise ValueError(f'direction must either be "inputs" or "outputs", but {direction} was given')
+
+        if direction == "inputs":
+            decoder_sequence_name = "past_sequence_length"
+            name = "past_key_values"
+        else:
+            decoder_sequence_name = "past_sequence_length + 1"
+            name = "present"
+
+        # generate tuples of (key, value, conv_state, ssm_state)
+        for i in range(self._normalized_config.num_layers):
+            inputs_or_outputs[f"{name}.{i}.key"] = {0: "batch_size", 1: decoder_sequence_name}
+            inputs_or_outputs[f"{name}.{i}.value"] = {0: "batch_size", 1: decoder_sequence_name}
+            inputs_or_outputs[f"{name}.{i}.conv_state"] = {0: "batch_size"}
+            inputs_or_outputs[f"{name}.{i}.ssm_state"] = {0: "batch_size"}
+
+    @property
+    def inputs(self) -> Dict[str, Dict[int, str]]:
+        common_inputs = {"input_ids": {0: "batch_size", 1: "sequence_length"}}
+        common_inputs["attention_mask"] = {0: "batch_size", 1: "sequence_length"}
+        common_inputs["position_ids"] = {0: "batch_size", 1: "sequence_length"}
+        self.add_past_key_values_states(common_inputs, direction="inputs")
+        return common_inputs
+
+    @property
+    def outputs(self) -> Dict[str, Dict[int, str]]:
+        common_outputs = {}
+        common_outputs["logits"] = {}
+
+        # outputs in an order of (key, value, conv_state, ssm_state)
+        for idx in range(self._normalized_config.num_layers):
+            common_outputs["key_cache.present.{}".format(idx)] = {}
+            common_outputs["value_cache.present.{}".format(idx)] = {}
+            common_outputs["conv_states.present.{}".format(idx)] = {}
+            common_outputs["ssm_states.present.{}".format(idx)] = {}
+
+        return common_outputs
```

Comment:
shoudln't this depends on the task ? (depending on the with-past suffix) also what about stateful support ?

Reply:
yeah probably needs to override the add_past_key_values method instead

Reply:
@rkazants